### PR TITLE
Use numeric, list, or data.frame in createColonies() & setLocation()

### PR DESCRIPTION
**Pull request by gregorgorjanc**
@janaobsteter have a look

### CODE REVIEW

File: R/Functions_L2_Colony.R [hunk 1]
```diff
@@ -1263,13 +1269,18 @@ collapse <- function(x) {
...
+                  sampleLocation = TRUE, sdLocation = NULL,
+                  simParamBee = NULL, ...) {
   if (is.null(simParamBee)) {
     simParamBee <- get(x = "SP", envir = .GlobalEnv)
   }
   if (is.null(p)) {
     p <- simParamBee$swarmP
   }
+  if (is.null(sdLocation)) {
+    sdLocation <- simParamBee$swarmSDLocation
```
Comment (janaobsteter) [inline R/Functions_L2_Colony.R:1282]: And this is 0 by default, right?

File: R/Functions_L2_Colony.R [hunk 2]
```diff
@@ -1304,12 +1315,23 @@ swarm <- function(x, p = NULL, year = NULL, nVirginQueens = NULL, simParamBee =
     #       https://github.com/HighlanderLab/SIMplyBee/issues/160
     tmp <- pullWorkers(x = x, nInd = nWorkersSwarm)
     currentLocation <- getLocation(x)
+    if (sampleLocation) {
+      if (length(sdLocation) != 2) {
```
Comment (janaobsteter) [inline R/Functions_L2_Colony.R:1319]: What if it was one number - the radius?

File: R/Functions_L2_Colony.R [hunk 3]
```diff
@@ -1733,58 +1757,91 @@ combine <- function(strong, weak) {
...
-      for (colony in seq_len(nCol)) {
-        if (!is.null(x[[colony]])) {
-          x[[colony]]@location <- location
+      } else if (is.list(location)) {
+        if (length(location) != n) {
+          stop("When argument location is a list, it must be of length n!")
         }
+        tmp <- sapply(X = location, FUN = length)
+        if (!all(tmp == 2)) {
+          stop("When argument location is a list, each list node must be of length 2!")
```
Comment (janaobsteter) [inline R/Functions_L2_Colony.R:1824]: And the length of the list should match the number of colonies

**Review by janaobsteter — COMMENTED** [review]
@gregorgorjanc , just a couple of minor comments/questions that you can implement or not

File: R/Functions_L2_Colony.R (hunk 1 above)
Comment (gregorgorjanc) [reply R/Functions_L2_Colony.R:1282]: Yes

File: R/Functions_L2_Colony.R (hunk 2 above)
Comment (gregorgorjanc) [reply R/Functions_L2_Colony.R:1319]: Hmm, at the moment we operate on (x,y) coordinates, and to sample a new (x,y) we need (dx,dy) where dx~N(0,sigma_{dx}^2) and dy~N(0,sigma_{dy}^2) so we need sigmas (SD) as input. This means that I did not need to implement the radius stuff at all;)

Having said this, someone might have idea about the radius but not the sigmas, which is what you are hinting at, correct? Maybe we provide a function to convert radius to sigmas? I guess also if we have a radius, then the two sigmas are equal, right? Is that really what you are asking? If so, can you copy this is a new spatial issue and we take it from there?

File: R/Functions_L2_Colony.R (hunk 3 above)
Comment (gregorgorjanc) [reply R/Functions_L2_Colony.R:1824]: Good catch! Will add this one! We actually do it above this code;)

File: R/Functions_L2_Colony.R (hunk 2 above)
Comment (janaobsteter) [reply R/Functions_L2_Colony.R:1319]: Yes, I was referring to sigmas being equal if we talk about a radius. Now discussed in #9

**Review by janaobsteter — APPROVED** [review]

**Comment by janaobsteter** [comment]
Merging this in

**Comment by gregorgorjanc** [comment]
Well done us - this is the 500-th PR!

**Pull request merged**